ansible/roles/ocp4-workload-authentication/tasks/workload.yml

```diff
@@ -1,48 +1,64 @@
 ---
-# Implement your Workload deployment tasks here
-
-- name: Check that ocp4_idm_install is defined
+- name: Set up ocp4_workload_authentication combined dictionary
+  set_fact:
+    ocp4_workload_authentication: >-
+      {{ ocp4_workload_authentication_defaults
+       | combine(ocp4_workload_authentication_input  | default( {} ),
+                 ocp4_workload_authentication_secret | default( {} ))
+      }}
+- name: Print combined role variables
+  debug: var=ocp4_workload_authentication
+
+- name: Check that ocp4_workload_authentication.idm_type is defined
   when:
-  - ocp4_idm_install is not defined
+  - ocp4_workload_authentication.idm_type is not defined
   fail:
-    msg: "ocp4_idm_install is not defined"
+    msg: "ocp4_workload_authentication.idm_type is not defined"
 
 - name: Setup HTPasswd Authentication
-  when: ocp4_idm_install == "htpasswd"
+  when: ocp4_workload_authentication.idm_type == "htpasswd"
   block:
-  # Generate Passwords if no passwords specified
   - name: Generate cluster admin password
-    when: ocp4_idm_htpasswd_admin_password | d('') | length == 0
+    when: ocp4_workload_authentication.htpasswd_admin_password | d('') | length == 0
```

Comment:
In your defaults for the role, this is set to `openshift_admin`, so wouldn't this always evaluate false and never generate?

Reply:
Maybe I'll read ahead next time.